crates/router/src/events/outgoing_webhook_logs.rs

```diff
@@ -0,0 +1,108 @@
+use api_models::{enums::EventType as OutgoingWebhookEventType, webhooks::OutgoingWebhookContent};
+use serde::Serialize;
+use serde_json::Value;
+use time::OffsetDateTime;
+
+use super::{EventType, RawEvent};
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub struct OutgoingWebhookEvent {
+    merchant_id: String,
+    event_id: String,
+    event_type: OutgoingWebhookEventType,
+    #[serde(flatten)]
+    content: Option<OutgoingWebhookEventContent>,
+    is_error: bool,
+    error: Option<Value>,
+    created_at_timestamp: i128,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+#[serde(tag = "outgoing_webhook_event_type", rename_all = "snake_case")]
+pub enum OutgoingWebhookEventContent {
+    Payment {
+        payment_id: Option<String>,
+        content: Value,
+    },
+    Refund {
+        payment_id: String,
+        refund_id: String,
+        content: Value,
+    },
+    Dispute {
+        payment_id: String,
+        attempt_id: String,
+        dispute_id: String,
+        content: Value,
+    },
+    Mandate {
+        payment_method_id: String,
+        mandate_id: String,
+        content: Value,
+    },
+}
+pub trait OutgoingWebhookEventMetric {
+    fn get_outgoing_webhook_event_type(&self) -> Option<OutgoingWebhookEventContent> {
+        None
+    }
+}
+impl OutgoingWebhookEventMetric for OutgoingWebhookContent {
+    fn get_outgoing_webhook_event_type(&self) -> Option<OutgoingWebhookEventContent> {
+        match self {
+            Self::PaymentDetails(payment_payload) => Some(OutgoingWebhookEventContent::Payment {
+                payment_id: payment_payload.payment_id.clone(),
+                content: masking::masked_serialize(&payment_payload).unwrap_or_default(),
```

Comment:
instead of default which is null,
can we add a json which tells us that serialization failed
e.g

```suggestion
                content: masking::masked_serialize(&payment_payload).unwrap_or(serde_json::json!({"error":"failed to serialize"})),
```